homeassistant/components/agent_dvr/camera.py

```diff
@@ -0,0 +1,223 @@
+"""Support for Agent camera streaming."""
+from datetime import timedelta
+import logging
+
+import voluptuous as vol
+
+from homeassistant.components.camera import SUPPORT_ON_OFF, SUPPORT_STREAM
+from homeassistant.components.mjpeg.camera import (
+    CONF_MJPEG_URL,
+    CONF_STILL_IMAGE_URL,
+    MjpegCamera,
+    filter_urllib3_logging,
+)
+from homeassistant.const import CONF_NAME
+from homeassistant.helpers import entity_platform
+from homeassistant.helpers.dispatcher import async_dispatcher_connect
+
+from .const import CAMERA_SCAN_INTERVAL_SECS, DOMAIN as AGENT_DOMAIN, SERVICE_UPDATE
+from .helpers import service_signal
+
+SCAN_INTERVAL = timedelta(seconds=CAMERA_SCAN_INTERVAL_SECS)
+
+_LOGGER = logging.getLogger(__name__)
+
+_DEV_EN_ALT = "enable_alerts"
+_DEV_DS_ALT = "disable_alerts"
+_DEV_EN_REC = "start_recording"
+_DEV_DS_REC = "stop_recording"
+_DEV_EN = "turn_on"
+_DEV_DS = "turn_off"
+_DEV_TOG = "toggle"
+_DEV_SNAP = "snapshot"
+
+CAMERA_SERVICES = {
+    _DEV_EN_ALT: "async_enable_alerts",
+    _DEV_DS_ALT: "async_disable_alerts",
+    _DEV_EN_REC: "async_start_recording",
+    _DEV_DS_REC: "async_stop_recording",
+    _DEV_EN: "async_turn_on",
+    _DEV_DS: "async_turn_off",
+    _DEV_TOG: "async_toggle",
+    _DEV_SNAP: "async_snapshot",
+}
+
+
+async def async_setup_entry(
+    hass, config_entry, async_add_entities, discovery_info=None
+):
+    """Set up the Agent cameras."""
+    filter_urllib3_logging()
+    cameras = []
+
+    for server in hass.data[AGENT_DOMAIN].values():
```

Comment:
```suggestion
    for server in hass.data[AGENT_DOMAIN][config_entry.entry_id].values():
```

Reply:
the agent connection objects are in an array inside hass.data[AGENT_DOMAIN] - using [config_entry.entry_id] breaks the integration

Reply:
There is going to be some refactoring required to make this work. Ideally each config entry is putting all of its data under `hass.data[AGENT_DOMAIN][config_entry.entry_id]` so it is clear which config entry it belongs to.

If you need to store multiple items per config entry, something like this could work:
```
hass.data[AGENT_DOMAIN][config_entry.entry_id] = {
    CONNECTION:  ...,
    SOMETHING_ELSE: ....
}
```